assets/app/scripts/services/apiEndpoint_v2.js

```diff
@@ -0,0 +1,311 @@
+                      // even though they are used for lookup.
+                      group: next.group,
+                      version: next.version
+                    };
+                  }
+                });
+                return memo;
+              }, {});
+    };
+
+    // temporary, until we get a new OPENSHIFT_CONFIG w/the new groupVersion structure
+    var mapLegacyAPis = function(pseudoGroups) {
+      return _.reduce(
+              pseudoGroups,
+              function(memo, next, i) {
+                // all old apis belong to the empty group, under v1
+                _.each(next.resources, function(resource, key) {
+                  memo['']['v1'][key] = {
+                    hostPort: next.hostPort,
+                    prefix: next.prefixes['v1'],
+                    group: '',
+                    version: 'v1'
+                  };
+                });
+                return memo;
+              }, {'':{'v1':{}}});
+    };
+
+    // TODO:
+    // - I temporarly have a hack in config.js to provide groupVersions
+    //   or not... which will trigger this switch in processing the objects.
+    //   The outcome is the same format, allowing for a single lookup.
+    //   The primary deviation in the code below is in urlForResource, where
+    //   it chooses lookup method based on the first argument, string vs object:
+    //   - DataService.get("pods", $routeParams.pod, context);
+    //   - DataService.get({resource:"jobs",group:"extensions",version:"v1beta1" }, $routeParams.job, context);
+    var apis = window.OPENSHIFT_CONFIG.api.openshift ?
+                // use the old config & transform to new map
+                mapLegacyAPis(window.OPENSHIFT_CONFIG.api) :
+                // use the new config & transform to the new map
+                mapApis(window.OPENSHIFT_CONFIG.groupVersions);
+
+    console.log('apis', apis);
+    // finds the api via spec, must have a full spec to work
+    var apiForResource = function(spec) {
+        return apis[spec.group] && apis[spec.group][spec.version] && apis[spec.group][spec.version][spec.resource];
+    };
+
+    // TODO: decide how to notify failure
+    var warnPartialSpec = function(spec) {
+      console.warn( 'Spec is missing ',
+                    _.difference(['group', 'version', 'resource'], _.keys(spec)),
+                    'api url cannot be constructed');
+    };
+
+    var findApiForResource = function(spec) {
+      return _.difference(
+                _.keys(spec),
+                ['group', 'version', 'resource']
+              ).length === 0 ?
+                apiForResource(spec) :
+                // NOTE:
+                // - we could support partial specs, but this is really not a good idea
+                // - inference of group could give completely different object type
+                // - inference of version could give unexpected data structure for the object
```

Comment:
I want a default version per group, like we have today... I don't want to scatter versions throughout the console controllers